addons/purchase/static/src/js/purchase_toaster_form_controller.js

```diff
@@ -0,0 +1,35 @@
+odoo.define('purchase.PurchaseToasterFormController', function (require) {
+"use strict";
+
+var core = require('web.core');
+var FormController = require('web.FormController');
+
+var _t = core._t;
+
+var PurchaseToasterFormController = FormController.include({
```

Comment:
it would probably be cleaner to extend button and not formcontroller..
isn't it possible to apply a widget on a button tag?

also you could use a text computed field to store in json the title and the content of the toaster, that'd prevent a new rpc .. not sure how i't  ll work with the html escaping though

and this rpc isn't done at the right time right? you'll display the toaster even if clicking the button resulted in a traceback